slsim/lens.py

```diff
@@ -670,18 +673,15 @@ def source_light_model_lenstronomy(self, band=None):
             self._source_type == "extended"
             or self._source_type == "point_plus_extended"
         ):
-            if self.light_profile == "single_sersic":
+            if self.source.light_profile == "single_sersic":
```

Comment:
is it perhaps possible to make this entire definition in the Source() class and providing the necessary inputs from the Lens() class?